### PR TITLE
refactor(index.js): reimplement signup and login actions

### PR DESCRIPTION
**Pull request by darrylyeo**
Replace calls to the old Firebase backend with HTTP POST calls to the new MongoDB backend.

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -1,83 +1,76 @@
...
-          callback();
-        }).catch(error => dispatch({ type: types.SIGN_UP_FAIL, error }));
-    }).catch((error) => {
-      dispatch({ type: types.SIGN_UP_FAIL, error });
+  axios
+    .post(`${API_PATH}/users/signup`, {
+      email: values.email,
+      password: values.password,
+    })
+    .then((response) => {
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: We should refactor all of the promise chaining into async/await
https://gyazo.com/0cb798a96ce4b4b15b50a09eb3913c99

File: src/actions/index.js [hunk 2]
```diff
@@ -1,83 +1,76 @@
...
 import {
   firebase,
   applicationsRef,
-  firestore,
   rsvpRef,
 } from '../config/firebase';
 
+const API_PATH = 'https://slohacks-backend-api.herokuapp.com';
+
 export const signUp = (values, callback) => (dispatch) => {
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: Let's make values more clear, destructure email and password out and fix below.

File: src/actions/index.js [hunk 3]
```diff
@@ -1,83 +1,76 @@
...
+      email: values.email,
+      password: values.password,
+    })
+    .then((response) => {
+      const userCredential = response.data;
+      dispatch({ type: types.SIGN_UP_GUCCI, userCredential });
+      callback();
+    })
+    .catch((error) => {
+      const { errorMessage } = error.response.data;
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: I actually have to fix the backend to send the errorMessage back.

File: src/actions/index.js [hunk 4]
```diff
@@ -1,83 +1,76 @@
...
+      dispatch({ type: types.SIGN_UP_FAIL, errorMessage });
     });
 };
 
 export const login = values => (dispatch) => {
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: Same as above, instead of values let's make it more specific

File: src/actions/index.js [hunk 5]
```diff
@@ -1,83 +1,76 @@
...
-        }).catch(() => {
-          dispatch({ type: types.UPDATE_APPLICATION_FALSE });
-          dispatch({ type: types.LOGIN_GUCCI, userCredential });
-        });
-      } else {
-        dispatch({ type: types.LOGIN_FAIL, error: { message: 'Email not verified, please verify your email.' } });
-      }
-    }).catch((error) => {
-      dispatch({ type: types.LOGIN_FAIL, error });
+  axios
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: Refactor the promise chaining

File: src/index.js [hunk 6]
```diff
@@ -1 +1 @@
-require('./index.jsx');
```
Comment (Jumer-Caragay) [inline src/index.js]: This should stay as single quote, The airbnb style guide is single quote.

File: src/actions/index.js [hunk 7]
```diff
@@ -95,15 +88,25 @@ export function submitResponse(formProps) {
...
+    storageRef
+      .put(resume)
+      .then(() => {
+        dispatch({ type: types.UPLOAD_RESUME_GUCCI, resume });
+        onChange(resume.name);
+      })
+      .catch(() => {
+        dispatch({
+          type: types.UPLOAD_RESUME_FAIL,
+          error: "File submitted isn't of type PDF or is too large.",
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: Ya make this single quotes

File: src/actions/index.js [hunk 8]
```diff
@@ -1,83 +1,76 @@
...
-      }
-    }).catch((error) => {
-      dispatch({ type: types.LOGIN_FAIL, error });
+  axios
+    .post(`${API_PATH}/users/login`, {
+      email: values.email,
+      password: values.password,
+    })
+    .then((response) => {
+      const userCredential = response.data;
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: console log the usercredential object, there is also a token we need to store. Look at auth reducer and see how we can improve the global state

File: src/actions/index.js [hunk 9]
```diff
@@ -1,133 +1,147 @@
...
+// import {
+//   firebase,
+//   applicationsRef,
+//   rsvpRef,
+// } from '../config/firebase';
 
-export const signUp = (values, callback) => (dispatch) => {
+const API_PATH = "https://slohacks-backend-api.herokuapp.com";
+
+export const signUp = (values, callback) => dispatch => {
```
Comment (toyoun) [inline src/actions/index.js]: Consider doing something about the `values` parameter. It's very vague.

Either (a) remove it and pass in `email` and `password` or (b) change the name to like `credentials`

File: src/actions/index.js [hunk 10]
```diff
@@ -1,133 +1,147 @@
...
+  // firebase
+  //   .auth()
+  //   .sendPasswordResetEmail(values.email)
+  //   .then((userCredential) => {
+  //     dispatch({ type: types.FORGOT_PASS_GUCCI, userCredential });
+  //     callback();
+  //   })
+  //   .catch((error) => {
+  //     dispatch({ type: types.FORGOT_PASS_FAIL, error });
+  //   });
```
Comment (toyoun) [inline src/actions/index.js]: Add axios calls to the /forgot-password route.

**Pull request closed without merging**